write-fonts/src/tables/post.rs

```diff
@@ -8,6 +10,38 @@ include!("../../generated/generated_post.rs");
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub struct PString(String);
 
+impl Post {
+    /// Construct a new version 2.0 table from a glyph order.
+    pub fn new_v2<'a>(order: impl IntoIterator<Item = &'a str>) -> Self {
+        let known_glyphs = crate::read::tables::post::DEFAULT_GLYPH_NAMES
+            .iter()
+            .enumerate()
+            .map(|(i, name)| (*name, i as u16))
+            .collect::<HashMap<_, _>>();
+        let mut name_index = Vec::new();
+        let mut storage = Vec::new();
+
+        for name in order {
+            match known_glyphs.get(name) {
+                Some(i) => name_index.push(*i),
+                None => {
+                    let idx = (known_glyphs.len() + storage.len()).try_into().unwrap();
```

Comment:
Not particularly germane to this until we get `> 16 bit` glyph ids, but I do think we should consider propagating errors rather than panicking wherever possible so that these can be reported in a friendly manner to the user. Nobody likes internal compiler errors :)

Reply:
Good observation... the pattern I've taken here is that there are separate validation and compilation passes; the validation pass ensures invariants are all upheld (and reports issues in aggregate) and then the compilation pass barrels through, assuming that the data is good.

I'm not totally sure this is the correct approach (for one thing, it encourages writing '//TODO: validate this' everywhere, but if we want to change that it will involve a general rearchitecting.